pkg/api/plugin.go

```diff
@@ -46,5 +51,7 @@ type Plugin interface {
 
 	HealthCheck(ctx context.Context, cs *OpenShiftManagedCluster) error
 
-	Update(ctx context.Context, cs *OpenShiftManagedCluster, azuredeploy []byte) error
+	// CreateOrUpdate either deploys or runs the update depending on the isUpdate argument
+	// this will call the PluginConfig.Deployer
+	CreateOrUpdate(ctx context.Context, cs *OpenShiftManagedCluster, azuredeploy []byte, isUpdate bool, Deployer DeployFn) error
```

Comment:
please open a follow up issue to replace `azuredeploy []byte` with `template map[string]interface{}` across the board?

Reply:
done https://github.com/openshift/openshift-azure/issues/547